docs/components/index.md

```diff
@@ -3,10 +3,17 @@
 This module contains classes that abstract various Karabo devices to make access
 easier.
 
+Component index:
+
+- [XGM][extra.components.XGM]
```

Comment:
Can we sort components alphabetically? 🙈

Reply:
To a degree yes :stuck_out_tongue: My thinking was that the component index should display the top-level components (i.e. not in a subsection like `Scans` or `Pulse patterns`) first, and then components in subsections. So I'd say that the top-level components and the ones in subsections can be ordered alphabetically, but the ordering should generally match what's displayed in the navigation bar on the left.

Reply:
Yes, but what I'm saying is that I'd prefer not to sort the components globally. e.g. if there are `Alpha` and `Zeta` top-level components I'd prefer the ordering to be:
- Alpha
- Zeta
- Pulse patterns...
- Scans...

Rather than:
- Alpha
- Pulse patterns...
- Scans...
- Zeta

Reply:
I find this confusing to be honnest. It might be fine as long as we don't have too many components, but then?
Or should we put all component in subsections (like `diagnostic` for XGM?) + an other flat page listing all components alphabetically?

Reply:
Sure, I don't feel very strongly about that. But I will probably leave it for another PR :)